engines/e_capi.c

```diff
@@ -1500,6 +1538,10 @@ static CAPI_KEY *capi_get_key(CAPI_CTX * ctx, const TCHAR *contname,
         CryptReleaseContext(key->hprov, 0);
         goto err;
     }
+    len = strlen(contname);
+    key->id = OPENSSL_malloc(len+1);
```

Comment:
Not coherent with next memcpy call, much more if it's compile with UNICODE enabled..
Malloc amount is in bytes, you pass strlen().